### PR TITLE
[Dec/Subplugins] Remove duplicated code

### PR DESCRIPTION
**Pull request by myungjoo**
Add "utility" internal API to remove
duplicated code.

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

### CODE REVIEW

File: ext/nnstreamer/tensor_decoder/meson.build [hunk 1]
```diff
@@ -23,7 +23,8 @@ static_library('nnstreamer_decoder_direct_video',
 
 # image labeling
 decoder_sub_image_labeling_sources = [
-  'tensordec-imagelabel.c'
+  'tensordec-imagelabel.c',
+  'tensordecutil.c'
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_decoder/meson.build:27]: Q: It's minor question but I just wonder what the file naming policy is here. `tensordecutil.c` or `tensordec-util.c` ?

Comment (myungjoo) [reply ext/nnstreamer/tensor_decoder/meson.build:27]: If we name it tensordec-util.c, it would look like a "util" subplugin for tensordec.

Comment (dongju-chae) [reply ext/nnstreamer/tensor_decoder/meson.build:27]: Right. Got it.

File: ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c [hunk 2]
```diff
@@ -230,68 +187,16 @@ bb_exit (void **pdata)
 {
   bounding_boxes *bdata = *pdata;
 
-  _free_labels (bdata);
+  _free_labels (&bdata->labeldata);
 
-  if (bdata->label_path)
-    g_free (bdata->label_path);
+  if (bdata->labeldata.label_path)
+    g_free (bdata->labeldata.label_path);
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c:193]: How about moving this line inside `_free_labels`?

File: ext/nnstreamer/tensor_decoder/tensordecutil.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  guint max_word_length; /**< The max size of labels */
+} imglabel_t;
+
+extern void
+loadImageLabels (imglabel_t *l);
+
+extern void
+initSingleLineSprite (singleLineSprite_t v, rasters_t r, uint32_t pv);
+
+extern void _free_labels(imglabel_t *data);
```
Comment (helloahn) [inline ext/nnstreamer/tensor_decoder/tensordecutil.h:38]: style check: `_free_labels(` -> `_free_labels (`

File: ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c (hunk 2 above)
Comment (myungjoo) [reply ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c:193]: No, we can't. Look at tensordecutil.c::35. label_path may be referred after calling it.

File: ext/nnstreamer/tensor_decoder/tensordecutil.h (hunk 3 above)
Comment (myungjoo) [reply ext/nnstreamer/tensor_decoder/tensordecutil.h]: Fixed.

File: ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c (hunk 2 above)
Comment (dongju-chae) [reply ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c:193]: Ah. I missed that. Got it.

Comment (zhoonit) [reply ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c:193]: Arguably, `label_path` shouldn't be part of this `imglabel_t`

1. `label_path` is not highly coupled to the label itself, rather it is a way used to inject data. There might be other way to initiate `imglabel_t` later on.
2. `label_path` always has to be treated separately to `imglabel_t` as seen in tensordecutil.c::35 or here.

File: ext/nnstreamer/tensor_decoder/meson.build (hunk 1 above)
Comment (wooksong) [inline ext/nnstreamer/tensor_decoder/meson.build:27]: How about making this src file be a common dep for all decoder sub-plugins?

Comment (myungjoo) [reply ext/nnstreamer/tensor_decoder/meson.build:27]: Image-segment and Directvideo don't use it yet.

File: ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c (hunk 2 above)
Comment (myungjoo) [reply ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c:193]: @zhoonit A good point. Let's refactor it.

Comment (myungjoo) [reply ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c]: Updated.

**Pull request merged**